toolchain/parse/node_kind.def

```diff
@@ -70,6 +70,30 @@ CARBON_PARSE_NODE_KIND_CHILD_COUNT(Name, 0,
 // A name in an expression context.
 CARBON_PARSE_NODE_KIND_CHILD_COUNT(NameExpression, 0, CARBON_TOKEN(Identifier))
 
+// ----------------------------------------------------------------------------
+
+// The comments below follow this pattern:
+
+// Descriptive heading:
+//   Child1
+//   Child2
+// Parent
+
+// In this, `Child1`, `Child2`, and `Parent` are all kinds of parse nodes, which
+// are then defined using the `CARBON_PARSE_NODE_KIND_*` macros. They are
+// written in the order they are emitted, and the indentation showing the tree
+// structure. Note that nodes are omitted in postorder (see
+// https://en.wikipedia.org/wiki/Tree_traversal#Depth-first_search), with the
```

Comment:
What do you think of removing this link? I think postorder traversals are a common term, and I think adding links everywhere it's mentioned stands to become noisy.

That said, this code isn't handling the emission, so you might consider pointing to a more authoritative source for how the code works, such as:

```
They are written in postorder. See tree.h for more information.
```

Note too, the tree supports both postorder and preorder traversal. While the default is to emit in postorder, you can pass `--preorder-parse-tree` and get preorder.